### PR TITLE
Default to not including the rails express patches in ruby derivations

### PR DESCRIPTION
**Pull request by njaremko**
###### Description of changes

Changes the default `ruby` derivations to not use the rails express patch set. They fundamentally change how ruby works in unexpected ways, and the current implementation breaks any attempt to build a ruby version that doesn't have a corresponding rails express patch set configured.

Prior to this change, you can't do this right now (fails due to patch sets being missing). After this change, this works fine.
```nix
ruby_3_0 = pkgs.mkRuby {
  version = pkgs.mkRubyVersion "3" "0" "5" "";
  sha256 = "sha256-mvxjgKAnpP4a4aPi7MtrSXucWsBjHBLKVvm3vrSEh3Y=";
};
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/ruby/default.nix [hunk 1]
```diff
@@ -91,10 +91,10 @@ let
         enableParallelBuilding = true;
 
         patches =
-          (import ./patchsets.nix {
+          if useRailsExpress then (import ./patchsets.nix {
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/ruby/default.nix:94]:
```suggestion
          lib.optionals useRailsExpress (import ./patchsets.nix {
```

File: pkgs/development/interpreters/ruby/default.nix [hunk 2]
```diff
@@ -91,10 +91,10 @@ let
...
             inherit patchSet useRailsExpress ops fetchpatch;
             patchLevel = ver.patchLevel;
-          }).${ver.majMinTiny}
+          }).${ver.majMinTiny} else []
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/ruby/default.nix:97]:
```suggestion
          }).${ver.majMinTiny}
```

**Comment by emptyflask** [comment]
Are the railsexpress patches even relevant these days? I can't find any benchmarks for using them with Ruby > 1.9.2, and presumably they're meant only for use with Rails...

Plus, using this virtually undocumented patchset that's maintained by one guy on github just seems like an unnecessary security risk.

**Comment by njaremko** [comment]
Is @marsam the only person that can approve this?

**Comment by michaelglass** [comment]
@njaremko looks like these changes are integrated into https://github.com/NixOS/nixpkgs/pull/218114

**Comment by njaremko** [comment]
Huh, that seems to be the case, guess I'll close this then...

**Pull request closed without merging**